gensim/models/word2vec.py

```diff
@@ -1819,18 +1820,31 @@ def predict_output_word(self, context_words_list, topn=10):
         if not hasattr(self.wv, 'vectors') or not hasattr(self, 'syn1neg'):
             raise RuntimeError("Parameters required for predicting the output words not found.")
 
-        word2_indices = [self.wv.get_index(w) for w in context_words_list if w in self.wv]
-        if not word2_indices:
-            logger.warning("All the input context words are out-of-vocabulary for the current model.")
-            return None
+        if all(isinstance(w, int) for w in context_words_list):
+            # then, indices were passed. Check they are valid
+            word2_indices = np.array(context_words_list)
+            if np.any(word2_indices < 0):
+                logger.warning("All input context word indices must be non-negative.")
+                return None
+            # take only the ones in the vocabulary
+            word2_indices = word2_indices[word2_indices < self.wv.vectors.shape[0]]
```

Comment:
```suggestion
            max_index = self.wv.vectors.shape[0]
            word2_indices = np.array([index for index in contex_words_list if 0 <= index < max_index])
```

Reply:
Does this change only aim at making things clearer ? Doesn't it slow the code down with this list operation instead of numpy-only operations ?

Reply:
Generally a project like Gensim is interested in 1st, correctness; 2nd, clarity; then 3rd, performance - and even then, mainly considering performance with respect to avoiding really-slow/wasteful approaches, or choosing equally-clear simple optimizations, or optimizing heavily-used code-paths that show up as being a problem in real-use-case profiling.

A single list-filter-iteration-via-a-list-comprehension is a very common Python idiom that should more-or-less be considered "costless" except for if-and-when you have special reason to have intense performance concerns about its likely uses. And here, since this method was motivated by the simulation of a word2vec "context window", and such windows are usually pretty small (default `window=5` means a 10-word inout, at most), juggling a tiny list is unlikely to be an important bottleneck. 

(But again: I suspect suggestions below may make this moot. And it's possible, if use of this method became common in ways we didn't expect – such as using giant input lists, or in a high-volume tight-loop that's key to some tasks' overall performance – it might become a concern. But those problems, if they materialize, might outgrow this method entirely.)